source/common/router/upstream_request.cc

```diff
@@ -224,6 +222,17 @@ void UpstreamRequest::cleanUp() {
   }
 }
 
+void UpstreamRequest::upstreamLog() {
+  Http::ResponseHeaderMap* upstream_headers = upstream_headers_ ? upstream_headers_.get() : nullptr;
```

Comment:
isn't upstream_headers.get() simply null if the pointer is not set

Reply:
done